source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -269,25 +269,21 @@ void Filter::parseV1Header(char* buf, size_t len) {
   }
 }
 
-bool Filter::parseExtensions(os_fd_t fd, uint8_t* buf, size_t buf_size, size_t* buf_off) {
+bool Filter::parseExtensions(Network::IoHandle& io_handle, uint8_t* buf, size_t buf_size,
+                             size_t* buf_off) {
   // If we ever implement extensions elsewhere, be sure to
   // continue to skip and ignore those for LOCAL.
   while (proxy_protocol_header_.value().extensions_length_) {
-    int bytes_avail;
-    auto& os_syscalls = Api::OsSysCallsSingleton::get();
-    if (os_syscalls.ioctl(fd, FIONREAD, &bytes_avail).rc_ < 0) {
-      throw EnvoyException("failed to read proxy protocol (no bytes avail)");
```

Comment:
Seems like we have some test coverage gaps here, this throw doesn't seem to have an associated test.

Reply:
We're still throwing the same error on line 284 on recv errors (tests were changed to throw recv errors instead of ioctl)